lib/Sema/TypeCheckType.cpp

```diff
@@ -2264,12 +2273,25 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
                  "@autoclosure");
         attrs.clearAttribute(TAK_autoclosure);
       }
-
+      
+      DifferentiabilityKind diffkind = DifferentiabilityKind::NonDifferentiable;
+      if (attrs.has(TAK_differentiable)) {
+        if (attrs.linear) {
+          // FIXME(bartchr): allow types to be marked
+          // as `@differentiable(linear)`.
+          diagnose(attrs.getLoc(TAK_differentiable),
+                   diag::linear_differentiable_type_disabled);
```

Comment:
```suggestion
                   diag::linear_differentiable_type_disabled);
          attrs.clearAttribute(TAK_differentiable);
```

Reply:
Since this is an error, we should clear the attribute as if it never existed and move on, like how Swift deals with the invalid `TAK_autoclosure` above.